### PR TITLE
[14.0][ADD]scrap_location_filter

### PR DESCRIPTION
**Pull request by Freni-OSI**
Ticket: #16625

### CODE REVIEW

**Comment by patrickrwilson** [comment]
@dreispt can you assist with reviewing this?

File: scrap_location_filter/models/stock.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    def _search(
+        self,
+        args,
+        offset=0,
+        limit=None,
+        order=None,
+        count=False,
+        access_rights_uid=None,
+    ):
+        context = self._context or {}
```
Comment (dreispt) [inline scrap_location_filter/models/stock.py]:
```suggestion
        context = self.env.context
```

File: scrap_location_filter/models/stock.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        if context.get("product_id") or context.get("lot_id"):
```
Comment (dreispt) [inline scrap_location_filter/models/stock.py]: ?
```suggestion
        if context.get("product_id") and context.get("lot_id"):
```

File: scrap_location_filter/models/stock.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                ]
+            )
+            locations = quants.mapped("location_id")
+            if not locations:
+                company_id = context.get("default_company_id") or self.env.company.id
+                warehouse = self.env["stock.warehouse"].search(
+                    [("company_id", "=", company_id)], limit=1
+                )
+                if warehouse:
+                    locations = warehouse.lot_stock_id
```
Comment (dreispt) [inline scrap_location_filter/models/stock.py]: On "else" locations will not be assigned, and we are in trouble.
I say just get rid of the "if"
```suggestion
                locations = warehouse.lot_stock_id
```

**Review by dreispt — COMMENTED** [review]
LGTM, made some minor suggestions

File: scrap_location_filter/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    "name": "Scrap Location Filter",
+    "version": "14.0.1.0.0",
+    "category": "Warehouse Management",
+    "license": "AGPL-3",
+    "depends": ["stock"],
+    "summary": "Filters scrap location",
+    "author": "Open Source Integrators, , Odoo Community Association (OCA)",
+    "website": "https://github.com/OCA/stock-logistics-warehouse",
+    "data": ["views/stock_views.xml"],
+    "maintainers": ["Open Source Integrators"],
```
Comment (rousseldenis) [inline scrap_location_filter/__manifest__.py]: Should be a GitHub login.

File: scrap_location_filter/README.rst [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+**Table of contents**
+
+.. contents::
+   :local:
+
+Usage
+=====
+
+- Go to Inventory > Operations > Scrap
+- Create a scarp order and select Location.
```
Comment (rousseldenis) [inline scrap_location_filter/README.rst]:
```suggestion
- Create a scrap order and select Location.
```

File: scrap_location_filter/models/stock.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+            args += [("id", "in", locations.ids)]
+        return super(StockLocation, self)._search(
+            args, offset, limit, order, count=count, access_rights_uid=access_rights_uid
+        )
+
+
+class StockScrap(models.Model):
+    _inherit = "stock.scrap"
```
Comment (rousseldenis) [inline scrap_location_filter/models/stock.py]: Could you split one model per file ?

File: scrap_location_filter/models/stock.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+    @api.onchange("product_id")
+    def _onchange_product_id(self):
+        self.location_id = False
```
Comment (rousseldenis) [inline scrap_location_filter/models/stock.py]: Is these behaviours wanted in this module ?

File: scrap_location_filter/tests/test_scrap_location_filter.py [hunk 8]
```diff
@@ -0,0 +1,59 @@
+# Copyright (C) 2021 - TODAY, Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo.tests.common import TransactionCase
+
+
+class StockScrapLocation(TransactionCase):
```
Comment (rousseldenis) [inline scrap_location_filter/tests/test_scrap_location_filter.py]: Change this to Savepointcase

File: scrap_location_filter/tests/test_scrap_location_filter.py [hunk 9]
```diff
@@ -0,0 +1,59 @@
+# Copyright (C) 2021 - TODAY, Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
```
Comment (rousseldenis) [inline scrap_location_filter/tests/test_scrap_location_filter.py]: You are mixing licenses through files. Moreover would you change to https url ?

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review

**Comment by patrickrwilson** [comment]
@OCA/community-maintainers 
Runboat isn't happy, is there someone who can take a look?
E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/r/rsync/rsync_3.1.3-8ubuntu0.1_amd64.deb  404  Not Found [IP: 91.189.88.142 80]
E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?

**Comment by rousseldenis** [comment]
> @OCA/community-maintainers Runboat isn't happy, is there someone who can take a look? E: Failed to fetch http://archive.ubuntu.com/ubuntu/pool/main/r/rsync/rsync_3.1.3-8ubuntu0.1_amd64.deb 404 Not Found [IP: 91.189.88.142 80] E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?

There is an open issue https://github.com/sbidoul/runboat/issues/48

That does not block for merges.

File: scrap_location_filter/models/__init__.py [hunk 10]
```diff
@@ -0,0 +1,5 @@
+# Copyright (C) 2021 Open Source Integrators
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
```
Comment (rousseldenis) [inline scrap_location_filter/models/__init__.py]: @Freni-OSI Could you fix all licensing LGPL => AGPL ?

**Review by patrickrwilson — APPROVED** [review]
LGTM

**Comment by rousseldenis** [comment]
@Freni-OSI You have still files with LGPL in header.

**Comment by patrickrwilson** [comment]
@rousseldenis I just fixed the licenses in tests, should be good now after checks.

**Comment by rousseldenis** [comment]
@Freni-OSI @patrickrwilson It seems that this is not working:

image

Tested on runboat

**Comment by Freni-OSI** [comment]
> @Freni-OSI @patrickrwilson It seems that this is not working:
> 
> image
> 
> Tested on runboat

Fixed.  It happened due to one of the previous suggestions. Thanks.

**Review by rousseldenis — APPROVED** [review]

**Comment by rousseldenis** [comment]
@dreispt Could you update your review ?

**Pull request merged**